src/org/opensolaris/opengrok/web/PageConfig.java

```diff
@@ -1353,31 +1361,35 @@ public String getSearchTitle() {
             title += req.getParameter("q") + " (full)";
         }
         if (req.getParameter(QueryBuilder.DEFS) != null && !req.getParameter(QueryBuilder.DEFS).isEmpty()) {
-            if (!title.isEmpty()) {
-                title += ", ";
-            }
+            title = addTitleDelimiter(title);
             title += req.getParameter(QueryBuilder.DEFS) + " (definition)";
         }
         if (req.getParameter(QueryBuilder.REFS) != null && !req.getParameter(QueryBuilder.REFS).isEmpty()) {
-            if (!title.isEmpty()) {
-                title += ", ";
-            }
+            title = addTitleDelimiter(title);
             title += req.getParameter(QueryBuilder.REFS) + " (reference)";
         }
         if (req.getParameter(QueryBuilder.PATH) != null && !req.getParameter(QueryBuilder.PATH).isEmpty()) {
-            if (!title.isEmpty()) {
-                title += ", ";
-            }
+            title = addTitleDelimiter(title);
             title += req.getParameter(QueryBuilder.PATH) + " (path)";
         }
         if (req.getParameter(QueryBuilder.HIST) != null && !req.getParameter(QueryBuilder.HIST).isEmpty()) {
-            if (!title.isEmpty()) {
-                title += ", ";
-            }
+            title = addTitleDelimiter(title);
             title += req.getParameter(QueryBuilder.HIST) + " (history)";
         }
 
-        // TODO: possibly add projects too
+        if (req.getParameterValues(QueryBuilder.PROJECT) != null && req.getParameterValues(QueryBuilder.PROJECT).length != 0) {
+            if (!title.isEmpty()) {
+                title += " ";
+            }
+            title += "in projects: ";
+            String projects[] = req.getParameterValues(QueryBuilder.PROJECT);
+            for (int i = 0; i < projects.length; i++) {
```

Comment:
Sorry that I didn't see this during the first review. In Java8 I think there is a possibility to make this shorter. Something like:
```java
Arrays.asList(a).stream().collect(Collectors.joining(","))
```